### PR TITLE
add kfp v2 compile a pipeline

### PR DESCRIPTION
**Pull request by connor-mccarthy**
Some outstanding content, such as cross-section links, to be added once merged into the `pipelines-v2-docs` branch.

### CODE REVIEW

**Comment by connor-mccarthy** [comment]
/cc @chensun
for content review

File: content/en/docs/components/pipelines/compile-a-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+```python
+cmplr.compile(addition_component, package_path='addition_component.yaml')
+```
+
+The `Compiler.compile` method accepts a few optional additional parameters:
+
+**`pipeline_name`** (`string`)
+
+Sets the name of the pipeline (or component). This is written to IR as the  `pipelineInfo.name` field. Will override the `name` passed to the `@dsl.pipeline` decorator.
```
Comment (chensun) [inline content/en/docs/components/pipelines/compile-a-pipeline.md:45]: nit: mention this name is used a pipeline context and explains a bit what is pipeline context?

Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/compile-a-pipeline.md:45]: By pipeline context, are you referring to the pipeline context manager we use to create pipeline spec? If so, isn't that largely an implementation detail and not something we need to explain to authors? If not, can let me know what you are referring to?

Comment (chensun) [reply content/en/docs/components/pipelines/compile-a-pipeline.md:45]: No, pipeline context is maybe a MLMD concept. It identifies a number of runs originated from the same pipeline definition. I think it provides a query scope for some features like caching and importer. Desmond and/or Jialu may provide better explanation on this.

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**